client/src/components/FeedbackPage/FeedbackTestimonialsSection/FeedbackTestimonialsSection.tsx

```diff
@@ -0,0 +1,159 @@
+  const theme = useAppSelector(selectTheme);
+  const { data, loading, error } = useAppSelector(state => state.feedback);
+  const [testimonials, setTestimonials] = useState<Testimonials>();
+
+  const testimonialsArray =
+    testimonials &&
+    Object.entries(testimonials).map(entry => ({
+      id: entry[0],
+      content: entry[1]
+    }));
+
+  useEffect(() => {
+    const testimonialData = data.reduce<Testimonials>(
+      ({ ratings, suggestions, recommendations, more }, curr): Testimonials => ({
+        [Testimonial.ratings]: {
+          title: 'How would you rate our app?',
+          list: [
+            ...ratings.list,
+            {
+              name: curr.name,
+              rating: curr.rating,
+              message: curr.reasons,
+              timestamp: curr.timestamp
+            }
+          ]
+        },
+        [Testimonial.suggestions]: {
+          title: 'Anything that can be improved?',
+          list: [
+            ...suggestions.list,
+            { name: curr.name, message: curr.suggestions, timestamp: curr.timestamp }
+          ]
+        },
+        [Testimonial.recommendations]: {
+          title: 'Would you recommend this app to someone else?',
+          list: [
+            ...recommendations.list,
+            { name: curr.name, message: curr.recommend, timestamp: curr.timestamp }
+          ]
+        },
+        [Testimonial.more]: {
+          title: 'Care to share more? ',
+          list: [...more.list, { name: curr.name, message: curr.more, timestamp: curr.timestamp }]
+        }
+      }),
+      {
+        [Testimonial.ratings]: { title: '', list: [] },
+        [Testimonial.suggestions]: { title: '', list: [] },
+        [Testimonial.recommendations]: { title: '', list: [] },
+        [Testimonial.more]: { title: '', list: [] }
+      }
+    );
+    setTestimonials(testimonialData);
+  }, [data]);
+
+  return (
+    <S.FeedbackTestimonialsSection id="testimonials">
+      <Container>
+        <Flex directionColumn>
+          <h2>Testimonials</h2>
+          {data.length > 0 ? (
+            <RequestDataWrapper data={data} loading={loading} error={error}>
+              {testimonialsArray?.map(({ id, content }, index) => (
+                <Accordion key={id} title={content.title} index={index}>
+                  {content.list
+                    .filter(item => item.message)
+                    .map(({ message, name, rating, timestamp }, i) => {
+                      const { day, year, time } = dateFormatter(timestamp, true);
+
+                      return (
+                        <S.FeedbackTestimonial key={`testimonial-${i + 1}`} themeType={theme}>
+                          <Flex directionColumn alignFlexStart>
+                            <S.FeedbackTestimonialTop>
+                              {rating && (
+                                <Flex justifyFlexStart>
+                                  {[...Array(5)].map((_, j) => (
```

Comment:
i'd recommend to extend `StarRating` component functionality with "view" mode and reuse this component here

Reply:
Could you explain more by what you mean "view mode"? And is this really necessary? Yes, code structures do resemble each other a bit, for instance this
```
<S.FeedbackTestimonialStarWrapper
    key={`star-${j + 1}`}
    active={rating >= j + 1}
    themeType={theme}
  >
    <IconStar />
  </S.FeedbackTestimonialStarWrapper>
```
looks similar to
```
<S.StarWrapper
    key={`star-${ratingValue}`}
    themeType={theme}
    active={ratingValue <= (ratingHovered! || rating!)}
  >
    <label htmlFor={id}>
      <S.Star
        name={id}
        value={ratingValue}
        checked={value === ratingValue}
        {...inputConfig}
        onClick={() => setRating(ratingValue)}
        onMouseEnter={() => setRatingHovered(ratingValue)}
        onMouseLeave={() => setRatingHovered(null)}
        {...otherProps}
      />
      <IconStar />
    </label>
  </S.StarWrapper>
````
but just by a little. And what concerns styles, I've reused them by creating a mixin `starWrapperRoot`.

Reply:
i mean that would be better to extend StarRating component to have readonly "state", with non-clickable stars but showing provided value. yes, i see you have mixin for wrapper and reuse IconStar component, but these code snippets really look very similar + updating one component functionality/styles in one place will be easier than looking for multiple places in app + in future it would be easier to export this component with full functionality to another project